0001-short-term-release-process/0001-release-process-short-term.md

```diff
@@ -0,0 +1,322 @@
+to have community users test something on an isolated network. We can branch it off `main` and
+discard the branch when it's done. The branch is intended to have a very short duration. It will be
+possible to apply fixes to it and do a new release on the same branch, in which case the pre-version
+specifier will be incremented. If the fixes are good, we can cherry pick the fix commits back into
+`main`.
+
+An owner should be designated to the whole experiment. They will produce the release, deploy it, and
+coordinate with users.
+
+#### Process
+
+* Prepare a light description that communicates the purpose of the release; we do not need a full
+  changelog for this type of release.
+* Create and checkout an `YYYY-MM-DD-alpha.1` branch from `main`.
+* Use `release-plz update` to bump crate versions.
+* Use a script to apply the `alpha` pre-release specifier to each bumped crate.
+* Create a `chore(release): alpha-YYYY-MM-DD` commit.
+* Push that branch to the upstream `maidsafe` repo, which will kick off the release workflow.
+* The release workflow will produce a public `pre-release` on Github, but the crates will *not* be
+  published.
+* Manually edit the Github Release to provide the description prepared in the first step.
+* Use the `Launch Network` workflow to deploy the `alpha` binaries to an isolated network.
+* Announce the availability of the binaries to the community. Users can use `safenode-manager`
+  and/or `safeup` with `--version` arguments to obtain the alpha binaries.
+* Perform testing
+* If a problem is identified, there is an opportunity for a small fix/test cycle:
+    - On the same alpha branch, apply the fix.
+    - Use `cargo release version alpha --package X --package Y` to increment the necessary crates to
+      `alpha.2`
+    - Deploy the fix either using an upgrade or by launching a new testnet
+    - Users can test
+    - Repeat if necessary (in practice we should not have many of these)
+* If need be, any fix commits applied on the alpha branch should be cherry picked into `main`.
+* The experiment is over and the branch should be deleted.
+
+The branch is discarded because we don't want the alpha version bumps back in `main`. Crates were
+also not published. A Github Release always creates a tag, so this will function as the historical
+record of the existence of the alpha release.
+
+### Release Candidates
+
+A release candidate (RC) is the binary that's intended to be released as a stable version. The set
+of features and fixes in the RC is what's included on `main` in the current cycle, i.e., between now
+and the last stable release. After about two weeks of development we should produce the RC for
+testing in the staging environment.
+
+An owner should be designated to the process, and it would be useful for this to cycle through
+everyone in the team. Once the RC branch is started, we won't accept new features on it, only fixes.
+Feature development can continue on `main`.
```

Comment:
- Is the`release candidate` expected to go out to community members or only `alpha` builds?
- I believe the intention is for release candidate to go out to the community, but the top summary on the `release candidate` release section doesn't make a note off this compared to the alpha release summary, though its briefly mentioned within the `Process` section (kind of buried) as :
>Invite users to participate in testing the RC. They can use safenode-manager to obtain the rc nodes and safeup for rc clients.


- Just to confirm, is a `release candidate` expected to run as an isolated network when asking the community to test it?

Reply:
I think `rc` goes out to community and would be initial integration into existing networks.

`alpha/beta` as @jacderida explained above is about experiments and separate networks (which may or may not go out to community. I think these correspond to what would be `DEV-01` eg testnets in the naming schemes

Reply:
The release candidate was definitely intended to go to the community. I will push a change to make that more clear in the first paragraph.

In my mind, I was thinking about the RC running on an isolated network, to be tested in isolation.

I would be interested to know what you think here Shu. If we are pushing it to the real network, would we add nodes to the existing network, or upgrade some current nodes? How would we verify our changes were working?

Should the three of us perhaps have a call to discuss this one? Just keen to reach a consensus and move the process forward as soon as possible.

Reply:
In a call we agreed that the RC would be tested in isolation.